doc/source/whatsnew/v1.3.0.rst

```diff
@@ -170,6 +170,60 @@ a copy will no longer be made (:issue:`32960`)
 The default behavior when not passing ``copy`` will remain unchanged, i.e.
 a copy will be made.
 
+.. _whatsnew_130.arrow_string:
+
+PyArrow backed string data type
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+We've enhanced the :class:`StringDtype`, an extension type dedicated to string data.
+(:issue:`39908`)
+
+It is now possible to specify a ``storage`` keyword option to :class:`StringDtype`, use
+pandas options or specify the dtype using ``dtype='string[pyarrow]'`` to allow the
+StringArray to be backed by a PyArrow array instead of a NumPy array of Python objects.
+
+The PyArrow backed StringArray requires pyarrow 1.0.0 or greater to be installed.
+
+.. warning::
+
+   ``string[pyarrow]`` is currently considered experimental. The implementation
+   and parts of the API may change without warning.
+
+.. ipython:: python
+
+   pd.Series(['abc', None, 'def'], dtype=pd.StringDtype(storage="pyarrow"))
+
+You can use the alias ``"string[pyarrow]"`` as well.
+
+.. ipython:: python
+
+   s = pd.Series(['abc', None, 'def'], dtype="string[pyarrow]")
+   s
+
+You can also create a PyArrow backed string array using pandas options.
+
+.. ipython:: python
+
+    with pd.option_context("string_storage", "pyarrow"):
+        s = pd.Series(['abc', None, 'def'], dtype="string")
+    s
+
+The usual string accessor methods work. Where appropriate, the return type of the Series
+or columns of a DataFrame will also have string dtype.
+
+.. ipython:: python
+
+   s.str.upper()
+   s.str.split('b', expand=True).dtypes
+
+String accessor methods returning integers will return a value with :class:`Int64Dtype`
+
+.. ipython:: python
+
+   s.str.count("a")
+
+See :ref:`text.types` for more.
```

Comment:
Is there actually more content about ArrowString in that doc page?

Reply:
not yet. updating text.rst will be done once we have the dtype sorted and have agreed on the array naming.  will open a draft PR shortly with the changes so far (even though it's fluid)  (and add the stuff removed from this PR https://github.com/pandas-dev/pandas/pull/39908#discussion_r642535386)

could remove this link till then.

Reply:
We shouldn't need to update any user facing documentation if using pyarrow can be implemented on StringArray itself.

Reply:
If there is not yet more content on the other page, I would just remove the link for now.

Reply:
done in https://github.com/pandas-dev/pandas/pull/39908/commits/ef02a435ec481eeac62fe00ebc87d501489e13fc